go/cmd/dolt/commands/diff.go

```diff
@@ -613,7 +619,41 @@ func diffRows(
 
 	defer rowIter.Close(sqlCtx)
 
-	err = writeDiffResults(sqlCtx, sch, unionSch, rowIter, rowWriter)
+	if dArgs.skinny {
+		modifiedColNames, err := getModifiedCols(sqlCtx, rowIter, unionSch, sch)
+		if err != nil {
+			return errhand.BuildDError("Error running diff query:\n%s", query).AddCause(err).Build()
+		}
+
+		// instantiate a new schema that only contains the columns with changes
+		var filteredUnionSch sql.Schema
```

Comment:
(minor/suggestion) It looks like you could put the conversion from column names to a new schema in the `getModifiedCols` function. Might be worth pushing that logic into the function (and probably tweaking the name) to keep this main section of code a little tighter/shorter. Not a big deal either way though, just a suggestion to consider.

Reply:
Hmm, I think I'll prefer to keep the code that converts the new schema in `diffRows` to keep `getModifiedCols` as simple as possible (ie. so that the function only does 1 thing). Let me know if you feel strongly about this!